isis/src/docsys/documents/DeveloperAddFeature/DeveloperAddFeature.xml

```diff
@@ -0,0 +1,187 @@
+              a template can be very useful.
+            </p>
+            <ol>
+                <li>Decide where your new application belongs</li>
+                <li>Create a directory using the same name as you new application</li>
+                <li>Create a "main.cpp" file</li>
+                <li>Create a "your_new_app.cpp" file</li>
+                <li>Create a "your_new_app.xml" file</li>
+                <li>Create any additional support files (e.g., support_class.cpp, support_class.h</li>
+            </ol>
+            <p>
+              Populate the new files with the code necessary to implement your feature.
+            </p>
+
+            <h3>Compile and link your new application for the first time</h3>
+            <p>
+              Since you are adding a new application to ISIS you will need to re-run the cmake command. This will add 
+              the new application to the list of applications the build system knows about.
+            </p>
+            <ol>
+              <li>Change you current working directory to the ISIS build directory (cd $ISISROOT)</li>
+              <li>Remove the contents of the build directory (rm -rf *)</li>
+              <li>Re-run the cmake command (cmake -GNinja ..)</li>
+              <li>Build the ISIS library and all of the applications (ninja)</li>
+            </ol>
+
+            <h3>Fix, run, hand test, debug the new application</h3>
+            <p>
+              Iterate over the following steps until you have your new feature producing the desired result 
+            </p>
+            <ol>
+              <li>Fix any compile or linking errors</li>
+              <li>Rebuild the ISIS library and your new application (ninja)</li>
+              <li>Run your application and test the user interface</li>
+              <li>Run your application and test the output</li>
+                <ol>
+                  <li>Cube DNs</li>
+                  <li>Cube labels</li>
+                  <li>History</li>
+                  <li>Original label</li>
+                  <li>print.prt file</li>
+                  <li>Cleanup any temporary files</li>
+                </ol>
+            </ol>
+
+            <h2><a name="Writing your new application">Writing your new application</a></h2>
+              Applications in ISIS are required to have tests that exercise the majority of the application specific code.
+              These test are written using Google Test (aka, gtest). The source code for all gtests is located in the "isis/tests" directory
+              starting at the root of the code repository. Create a new file for your new application with the name "FunctionalTestsYournewapp.cpp", and
+              write tests to exercise the application. Remember to test only the functionality of your new code and not the ISIS library. Tests for the
+              ISIS library already exist.
```

Comment:
Mention that you should unit test any support classes that are sufficiently complex.

Reply:
Also link to the testing docs here